internal/resources/azure/virtual_network_peering.go

```diff
@@ -0,0 +1,131 @@
+package azure
+
+import (
+	"fmt"
+
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+// VirtualNetworkPeering struct represents a VNET peering.
+//
+
+// Resource information: https://azure.microsoft.com/en-us/services/virtual-network/#overview
+// Pricing information: https://azure.microsoft.com/en-us/pricing/details/virtual-network/
+type VirtualNetworkPeering struct {
+	Address           string
+	SourceRegion      string
+	DestinationRegion string
+	SourceZone        string
+	DestinationZone   string
+
+	DataProcessedGB *float64 `infracost_usage:"data_processed_gb"`
+}
+
+var VirtualNetworkPeeringUsageSchema = []*schema.UsageItem{
+	{Key: "data_processed_gb", DefaultValue: 0, ValueType: schema.Float64},
+}
+
+func (r *VirtualNetworkPeering) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+func (r *VirtualNetworkPeering) BuildResource() *schema.Resource {
+	costComponents := []*schema.CostComponent{
+		r.egressDataProcessedCostComponent(),
+		r.ingressDataProcessedCostComponent(),
```

Comment:
We may want to swap them places, so Inbound goes first how it is listed in issue details.
```suggestion
		r.ingressDataProcessedCostComponent(),
		r.egressDataProcessedCostComponent(),
```